kv/memdb/memory_mutation_zkevm.go

```diff
@@ -0,0 +1,78 @@
+package memdb
+
+import (
+	"github.com/gateway-fm/cdk-erigon-lib/kv"
+	"fmt"
+)
+
+func (m *MemoryMutation) Snapshot(id uint64) error {
+	m.snapshotsMutex.Lock()
+	defer m.snapshotsMutex.Unlock()
+
+	if _, exists := m.snapshots[id]; exists {
+		return fmt.Errorf("snapshot with id %d already exists", id)
+	}
+	snapshot, err := m.Clone(m.db, m.tmpDir, m)
+	if err != nil {
+		return err
+	}
+	m.snapshots[id] = snapshot
+	return nil
+}
+
+func (m *MemoryMutation) SnapshotFlush(id uint64, tx kv.RwTx) error {
+	m.snapshotsMutex.Lock()
+	snapshot, exists := m.snapshots[id]
+	if exists {
+		delete(m.snapshots, id)
+	}
+	m.snapshotsMutex.Unlock()
+
+	if !exists {
+		return fmt.Errorf("snapshot with id %d does not exist", id)
+	}
+	return snapshot.Flush(tx)
```

Comment:
maybe here we want to attempt the flush and check for an error before removing the snapshot from the map?  We might want to retry on transient errors for example